Tools/Cli-LoRa-Device-Provisioning/Program.cs

```diff
@@ -0,0 +1,543 @@
+                "deveui",
+                Required = true,
+                HelpText = "DevEUI / Device Id.")]
+            public string DevEui { get; set; }
+        }
+
+        static int Main(string[] args)
+        {
+            WriteAzureLogo();
+            Console.WriteLine("Azure IoT Edge LoRaWAN Starter Kit LoRa Leaf Device Provisioning Tool.");
+            Console.Write("This tool complements ");
+            Console.ForegroundColor = ConsoleColor.Blue;
+            Console.WriteLine("http://aka.ms/lora");
+            Console.ResetColor();
+            Console.WriteLine();
+
+            var success = Parser.Default.ParseArguments<ListOptions, QueryOptions, VerifyOptions, AddOptions, UpdateOptions, RemoveOptions>(args)
+                .MapResult(
+                    (ListOptions opts) => RunListAndReturnExitCode(opts),
+                    (QueryOptions opts) => RunQueryAndReturnExitCode(opts),
+                    (VerifyOptions opts) => RunVerifyAndReturnExitCode(opts),
+                    (AddOptions opts) => RunAddAndReturnExitCode(opts),
+                    (UpdateOptions opts) => RunUpdateAndReturnExitCode(opts),
+                    (RemoveOptions opts) => RunRemoveAndReturnExitCode(opts),
+                    errs => false);
+
+            if ((bool)success)
+            {
+                Console.ForegroundColor = ConsoleColor.Green;
+                Console.WriteLine();
+                Console.WriteLine("Successfully terminated.");
+                Console.ResetColor();
+
+                return (int)ExitCode.Success;
+            }
+            else
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine();
+                Console.WriteLine("Terminated with errors.");
+                Console.ResetColor();
+
+                return (int)ExitCode.Error;
+            }
+        }
+
+        private static object RunListAndReturnExitCode(ListOptions opts)
+        {
+            int page;
+            int total;
+
+            if (!configurationHelper.ReadConfig())
+                return false;
+
+            if (!int.TryParse(opts.Page, out page))
+                page = 10;
+
+            if (!int.TryParse(opts.Total, out total))
+                total = -1;
+
+            var isSuccess = iotDeviceHelper.QueryDevices(configurationHelper, page, total).Result;
+
+            Console.WriteLine();
+            Console.WriteLine("Done.");
+            return isSuccess;
+        }
+
+        private static object RunQueryAndReturnExitCode(QueryOptions opts)
+        {
+            if (!configurationHelper.ReadConfig())
+                return false;
+
+            var twin = iotDeviceHelper.QueryDeviceTwin(opts.DevEui, configurationHelper).Result;
+
+            if (twin != null)
+            {
+                var twinData = JObject.Parse(twin.Properties.Desired.ToJson());
+                twinData.Remove("$metadata");
+                twinData.Remove("$version");
```

Comment:
This is repetitive. You could add a method:

string TwinToString(TwinCollection collection)
{
          var twinData = JObject.Parse(collection.ToJson());
          twinData.Remove("$metadata");
          twinData.Remove("$version");
          return twinData.ToString();
}

Reply:
done!